### PR TITLE
762 and 781 and 775

### LINKED ISSUE #762: Make all new species split from existing ones
Instead of being randomly generated.

### LINKED ISSUE #775: Fix nitrogenase offset in editor
As discussed in this thread:

https://forum.revolutionarygamesstudio.com/t/tooltips-update-for-0-4-2/605

Nein99 is working on updating the GUI at the moment and I can help out with the tooltips to learn more about how the UI works.

### LINKED ISSUE #781: The process list update needs cleanup
There's now a few issues after https://github.com/Revolutionary-Games/Thrive/pull/780 left to resolve:
- [x] When the player dies the process list isn't reset, also when a cell finishes splitting it isn't done. So the recreation of the process list should be added to the resetOrganelleLayout function.
- [x] There are two copies of the process list generation code. They should be merged

### PR DESCRIPTION
**Pull request by Untrustedlife**
Creating draft pull request just so its easier for me to look at the diff
fixes #762 
fixes #781 
fixes #775

### CODE REVIEW

File: scripts/microbe_stage/microbe_editor/microbe_editor.as [hunk 1]
```diff
@@ -125,15 +125,28 @@ class MicrobeEditor{
             playerSpecies.organelles);
 
         editedMicrobe.resize(0);
+        playerSpecies.stringCode="";
         for(uint i = 0; i < templateOrganelles.length(); ++i){
             editedMicrobe.insertLast(cast<PlacedOrganelle>(templateOrganelles[i]));
+            playerSpecies.stringCode+=cast<PlacedOrganelle>(templateOrganelles[i])._organelle.gene;
```
Comment (hhyyrylainen) [inline scripts/microbe_stage/microbe_editor/microbe_editor.as]: The cast here should only be done once as it is (very) expensive.

Comment (Untrustedlife) [reply scripts/microbe_stage/microbe_editor/microbe_editor.as]: Okay, will fix that

Comment (Untrustedlife) [reply scripts/microbe_stage/microbe_editor/microbe_editor.as]: fixed

**Comment by Untrustedlife** [comment]
All relavant issues have been completed

**Comment by Untrustedlife** [comment]
> There's a bunch of places where for efficiency's sake I'd like to see `const` and `const &in` parameters used (instead of plain handles that cause atomic reference counting operations and strings by value which will probably copy the whole thing each time the function is called)

Can you explain more, where you want these changes.

**Pull request merged**